tests/test_serialize.cc

```diff
@@ -449,4 +449,22 @@ TEST(test_serialize, test_joint_streamable) {
   oss << dist;
 }
 
+TEST(test_serialize, test_ThreadPool) {
```

Comment:
nit: might be worth a `nullptr` test as well

Reply:
Will do!

Reply:
I expanded test coverage substantially, have a look.